azure-pipelines.yml

```diff
@@ -3,20 +3,102 @@
 # Add steps that build, run tests, deploy, and more:
 # https://aka.ms/yaml
 
+name: marain_ps_deployment
+
 trigger:
-- master
-
-pool:
-  vmImage: 'ubuntu-latest'
-
-steps:
-- task: AzurePowerShell@5
-  displayName: 'Deploy dev environment'
-  inputs:
-    azureSubscription: '$(Endjin_AzureServiceConnection)'
-    ScriptType: 'FilePath'
-    ScriptPath: 'Solutions/Marain.Instance.Deployment/Deploy-MarainInstanceInfrastructure.ps1'
-    ScriptArguments: '-AzureLocation $(Endjin_AzureLocation) -EnvironmentSuffix $(Endjin_EnvironmentSuffix) -AadTenantId $(Endjin_AadTenantId) -SubscriptionId $(Endjin_SubscriptionId) -InstanceManifest $(Build.SourcesDirectory)/Solutions/InstanceManifests/Development.json -AadAppIds @{"mardevtenancy"="$(Endjin_TenancyAppId)";"mardevoperationscontrol"="$(Endjin_OperationsControlAppId)"} -DoNotUseGraph'
-    preferredAzurePowerShellVersion: '2.6.0'
-    pwsh: true
-    workingDirectory: 'Solutions/Marain.Instance.Deployment'
\ No newline at end of file
+  - feature/powershell-deploy
+
+stages:
+- stage: Build
+  displayName: Build stage
+  jobs:
+  - job: Build
+    displayName: Build
+    pool:
+      vmImage: 'ubuntu-latest'
+    steps:
+    - publish: $(Build.SourcesDirectory)/Solutions
+      artifact: marain_instance
+
+
+- stage: deploy_to_jd
```

Comment:
Is this a temporary thing, or are we expecting to have one stage in this file for each distinct environment we deploy to?

Reply:
Yes, exactly that - 1 stage per environment.

Ultimately, I think we'll need to move our deployment pipeline out of this repo and into our own deployment configuration repo (I'm working on a simple template repo that can be cloned to get people up-and-running - https://github.com/marain-dotnet/Marain.Deployment)